.github/workflows/check-odo.yml

```diff
@@ -13,33 +13,59 @@ jobs:
     steps:
       - name: Check Out Code
         uses: actions/checkout@v4
-      - name: Get latest ODO version
+      - name: Get latest ODO nightly build version
         run: |
           echo "REPO_ODO_VERSION=$(cat src/tools.json | jq -r .odo.version)" >> $GITHUB_ENV
-          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName,url)
-          echo "LATEST_TOOL_RELEASE=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName | sed 's|v||')" >> $GITHUB_ENV
-          echo "LATEST_TOOL_URL=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .url)" >> $GITHUB_ENV
+          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName)
+          latestReleaseTag="$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName)"
+          echo "LATEST_TOOL_RELEASE_TAG=${latestReleaseTag}" >> $GITHUB_ENV
+          echo "LATEST_TOOL_RELEASE_VERSION=$(echo ${latestReleaseTag} | sed 's|v||')" >> $GITHUB_ENV
+          commitId="$(git ls-remote --refs --heads https://github.com/${{ env.TOOL_REPO }} refs/heads/main 2>/dev/null | cut -c1-9 2>/dev/null)"
+          echo "LATEST_TOOL_COMMIT_ID=${commitId}" >> $GITHUB_ENV
+          echo "LATEST_TOOL_NIGHTLY_BUILD=${commitId}-nightly" >> $GITHUB_ENV
       - name: Find existing PR for ODO version
         run: |
-          echo PR_EXISTS=$(gh pr --repo ${{ github.repository }} list --state all --search "update odo ${{env.LATEST_TOOL_RELEASE}} in:title" --json url | jq length) >> $GITHUB_ENV
+          latestReleaseVersionString="${{ env.LATEST_TOOL_RELEASE_VERSION }} (${{ env.LATEST_TOOL_NIGHTLY_BUILD }})"
+          echo "LATEST_TOOL_RELEASE_VERSION_STRING=${latestReleaseVersionString}" >> $GITHUB_ENV
+          echo PR_EXISTS=$(gh pr --repo ${{ github.repository }} list --state open --search "update odo cli to ${latestReleaseVersionString} in:title" --json url | jq length) >> $GITHUB_ENV
       - name: Update src/tools.json with latest odo version
-        if: ${{ (env.LATEST_TOOL_RELEASE != '') && (env.LATEST_TOOL_RELEASE != env.REPO_ODO_VERSION) && (env.PR_EXISTS == 0) }}
+        if: ${{ (env.LATEST_TOOL_NIGHTLY_BUILD != '') && (env.LATEST_TOOL_RELEASE_VERSION_STRING != env.REPO_ODO_VERSION) && (env.PR_EXISTS == 0) }}
         run: |
-          jq --indent 4 '.odo.version = "${{ env.LATEST_TOOL_RELEASE }}"' src/tools.json | jq --indent 4 '.odo.versionRange = "^${{ env.LATEST_TOOL_RELEASE }}"' | jq --indent 4 '.odo.versionRangeLabel = "version >= ${{ env.LATEST_TOOL_RELEASE }}"' > src/tools.json.new
+          jq --indent 4 '.odo.version = "${{env.LATEST_TOOL_RELEASE_VERSION_STRING}}"' src/tools.json \
+           | jq --indent 4 '.odo.versionRange = "${{env.LATEST_TOOL_RELEASE_VERSION_STRING}}"' \
+           | jq --indent 4 '.odo.versionRangeLabel = "version == ${{env.LATEST_TOOL_RELEASE_VERSION_STRING}}"' > src/tools.json.new
```

Comment:
I don't think `semver` can handle strings of the form : `3.15.0 (6d83e4043-nightly)`. I think we need to keep this as 3.15.0.

Reply:
I don't remember for sure, but I think we do not use semver with these `versionRange` and `versionRangeLabel` properties on with any others when installing the binaries

Reply:
It looks like we do at https://github.com/redhat-developer/vscode-openshift-tools/blob/b80a77ac7a4dfbf11a55980e2eed463c7c11a022/src/tools.ts#L99 if you look up the chain.

Reply:
And I can't leave it just as `3.15.0` because otherwise it will not make any difference between builds. Neither construct it as `3.15.0. 6d83e4043` because `6d83e4043` is not an ordinal build number - it's a part of commit ID (a quite unique number, but semver won't able to work property with it)

Anyway, using `3.15.0 (6d83e4043-nightly)` in `version`, `versionRange` and `versionRangeLabel` do not result into any problems, as far as I know.

Reply:
> semver.satisfies(await ToolsConfig.getVersion(location), versionRange))

:disappointed:  You're right... But then I really cannot construct a valuable ordinal build number as version build number never changes for nightly builds, so  for this release it always stays `3.15.0` and even if I construct it as `previousReleaseVersion +1` (=== 3.15.1?)  it will fail updating it on the second nightly build created because  its 'constructed version number will be `3.15.1` too.

Need to think what can we do with the version/versionRange here

Reply:
In order to create a valuable and comparable label, I believe, it's enough to get a date/time of commit by its ID.
The problem is that I cannot get this info from a remote git repository (using `gh repo ... `  and `git ls-remote ...` commands.
Any thoughts?

Reply:
The question with `semver.satisfies()` is resolved: now the nightly version is changed to be in <Last Release Version + 0.0.1>-<Build Number>-nightly, so, f.i., for the release 3.15.0 and the commit with ID 6d83e4043 the nightly version will be generated as 3.15.1-116-nightly, which is fully semver compatible and when comparing:

3.15.0 < 3.15.1-116-nightly < 3.15.1